### DISCUSSION TITLE
What's the difference between Language Server and Language Service?

### DISCUSSION THREAD

**Discussion by liaodalin19903** [microsoft/vscode-discussions#1358] [Extension Development QnA] [answered]
When I read the Language Extension :

>Programmatic Language Features include auto completion, error checking, and jump to definition. These features are often powered by a **Language Server**, a program that analyzes your project to provide the dynamic features. One example is the [typescript-language-features](https://github.com/microsoft/vscode/tree/main/extensions/typescript-language-features) extension bundled in VS Code. It utilizes the [TypeScript **Language Service**](https://github.com/microsoft/TypeScript/wiki/Using-the-Language-Service-API) to offer Programmatic Language Features such as:

What's the difference between Language Server and Language Service?

**Comment by olguzzar** [comment, marked as answer]
Check out [this documentation on embedded languages](https://code.visualstudio.com/api/language-extensions/embedded-languages) that explains the relationship between a language service and language server. It also links to a sample extension so you can see how that translates into code.

Here's a relevant snippet from the docs:

> A language service is a library that implements [programmatic language features](https://code.visualstudio.com/api/language-extensions/programmatic-language-features) for a single language. A language server can embed language services to handle embedded languages.
